File: frontend/src/types/tax.ts
export interface TaxLot {
  id: number;
  tradingsymbol: string;
  exchange: string;
  buy_date: string;
  buy_price: number;
  quantity: number;
  remaining_quantity: number;
  sell_date: string | null;
  sell_price: number | null;
  realized_pnl: number | null;
  holding_type: string;
  created_at: string;
}

export interface TaxSummary {
  total_stcg: number;
  total_ltcg: number;
  total_intraday: number;
  total_fno: number;
  estimated_stcg_tax: number;
  estimated_ltcg_tax: number;
  fy: string;
}

export interface DailyTaxEstimate {
  date: string;
  stcg_to_date: number;
  ltcg_to_date: number;
  intraday_to_date: number;
  fno_to_date: number;
  estimated_tax: number;
  advance_tax_due: number;
}

export interface WashSale {
  tradingsymbol: string;
  sell_date: string;
  sell_price: number;
  rebuy_date: string;
  rebuy_price: number;
  loss_amount: number;
}

export interface TaxComputeResponse {
  lots_created: number;
  lots_updated: number;
  message: string;
}

File: app/schemas/tax.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class TaxLotResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    tradingsymbol: str
    exchange: str
    buy_date: datetime
    buy_price: float
    quantity: int
    remaining_quantity: int
    sell_date: datetime | None
    sell_price: float | None
    realized_pnl: float | None
    holding_type: str
    created_at: datetime


class TaxSummaryResponse(BaseModel):
    total_stcg: float
    total_ltcg: float
    total_intraday: float
    total_fno: float
    estimated_stcg_tax: float
    estimated_ltcg_tax: float
    fy: str


class DailyTaxEstimate(BaseModel):
    date: date
    stcg_to_date: float
    ltcg_to_date: float
    intraday_to_date: float
    fno_to_date: float
    estimated_tax: float
    advance_tax_due: float


class WashSaleResponse(BaseModel):
    tradingsymbol: str
    sell_date: datetime
    sell_price: float
    rebuy_date: datetime
    rebuy_price: float
    loss_amount: float


class TaxComputeResponse(BaseModel):
    lots_created: int
    lots_updated: int
    message: str
